common/djangoapps/track/backends/oli.py

```diff
@@ -0,0 +1,121 @@
+"""OLI analytics service event tracker backend."""
+
+from __future__ import absolute_import
+
+import json
+import logging
+
+from requests_oauthlib import OAuth1Session
+
+from django.contrib.auth.models import User
+
+from student.models import anonymous_id_for_user
+from track.backends import BaseBackend
+
+
+LOG = logging.getLogger(__name__)
+
+
+class OLIAnalyticsBackend(BaseBackend):
+    """
+    Transmit events to the OLI analytics service
+    """
+    def __init__(self, **kwargs):
+        super(OLIAnalyticsBackend, self).__init__(**kwargs)
+
+        # the oli analytics service endpoint will be at
+        # "{}{}".format(oli_analytics_service_url, path)
+        self.url = kwargs.get('url', None)
+        self.path = kwargs.get('path', None)
+
+        # shared key for OLI analytics service
+        self.key = kwargs.get('key', None)
+
+        # shared secret for OLI analytics service
+        self.secret = kwargs.get('secret', None)
+
+        self.oauth = OAuth1Session(self.key, client_secret=self.secret)
+
+        # only courses with id in this set will have their data sent
+        self.course_id_set = set(kwargs.get('course_ids', []))
```

Comment:
@stvstnfrd 
A lot of these lines were not from me (see previous 2 commits) so I'm not sure why the variables were named that way.
I will change it to be consistent here. 
The timeout_in_sec variable is actually not used so I removed it.